setup.py

```diff
@@ -93,6 +90,11 @@
             'python-neutronclient',
             'raven',
             'requests != 2.13.0',
+        ],
+        'test': [
+            'tox',
+            'pip-tools',
+            'pytest',           # for tox.ini
```

Comment:
Does it worth to add mock==2.0.0, fudge, pytest-cov==1.6, coverage==3.7.1 and something other from the .tox.ini here too?

Reply:
not sure which is better -- to document the dependencies in `setup.py` or in `tox.ini`. IMHO, the ones specified by `setup.py` are install dependencies which are not necessarily the dependencies for running tests, which are used by developers only. and requirements.txt is the right place for them. but in our case, requirements.txt is generated. so, `tox.ini` is good enough. i think.

Reply:
If so, why do you include pytest in setup.py?
From other side, I usually run pytest without tox when fixing them and debugging, so I have to manually install mock, flake8 and the rest.

Reply:
@kshtsk it was included by "install_requires". please see my commit message

> setup.py: move test related stuff to optional deps

i intended to move them instead of removing them.